server/thw/thw.go

```diff
@@ -194,3 +203,49 @@ func (hw *HashWheel) GetNextExpiration(before int64) int64 {
 	}
 	return math.MaxInt64
 }
+
+// AppendEncode writes out the contents of the THW into a binary snapshot
+// and returns it. The high seq number is included in the snapshot and will
+// be returned on decode.
+func (hw *HashWheel) Encode(highSeq uint64) []byte {
+	b := make([]byte, 0, 17+(hw.count*(2*binary.MaxVarintLen64)))
+	b = append(b, 1)                                  // Magic version
+	b = binary.LittleEndian.AppendUint64(b, hw.count) // Entry count
+	b = binary.LittleEndian.AppendUint64(b, highSeq)  // Stamp
+	for _, slot := range hw.wheel {
+		for v, ts := range slot.entries {
+			b = binary.AppendVarint(b, ts)
+			b = binary.AppendUvarint(b, v)
+		}
+	}
+	return b
+}
+
+// Decode snapshots a binary-encoded THW and replaces the contents of this
+// THW with them. Returns the high seq number from the snapshot.
+func (hw *HashWheel) Decode(b []byte) (uint64, error) {
+	if len(b) < 17 {
+		return 0, io.ErrShortBuffer
+	}
+	if b[0] != 1 {
+		return 0, fmt.Errorf("unknown magic %d", b[0])
```

Comment:
Add to know errors?

Reply:
Done.

Reply:
nvm I see its variable. But we could make it a fixed error.